src/approaches/base_approach.py

```diff
@@ -9,6 +9,7 @@
     ParameterizedOption, Action, Dataset, Metrics, InteractionRequest, \
     InteractionResult
 from predicators.src.settings import CFG
+from predicators.src.utils import ExceptionWithInfo
```

Comment:
Let's make `EnvironmentFailure` in base_env.py also inherit from ExceptionWithInfo, with dictionary key "offending_objects"?

Reply:
done